codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/IdempotencyTokenProviderCustomization.kt

```diff
@@ -15,25 +16,42 @@ import software.amazon.smithy.rust.codegen.core.smithy.customize.NamedCustomizat
 /**
  * Add a `make_token` field to Service config. See below for the resulting generated code.
  */
-class IdempotencyTokenProviderCustomization : NamedCustomization<ServiceConfig>() {
+class IdempotencyTokenProviderCustomization(private val runtimeMode: SmithyRuntimeMode) : NamedCustomization<ServiceConfig>() {
+
     override fun section(section: ServiceConfig): Writable {
         return when (section) {
             is ServiceConfig.ConfigStruct -> writable {
-                rust("pub (crate) make_token: #T::IdempotencyTokenProvider,", RuntimeType.IdempotencyToken)
+                if (runtimeMode.defaultToMiddleware) {
+                    rust("pub (crate) make_token: #T::IdempotencyTokenProvider,", RuntimeType.IdempotencyToken)
+                }
             }
 
             ServiceConfig.ConfigImpl -> writable {
-                rust(
-                    """
-                    /// Returns a copy of the idempotency token provider.
-                    /// If a random token provider was configured,
-                    /// a newly-randomized token provider will be returned.
-                    pub fn make_token(&self) -> #T::IdempotencyTokenProvider {
-                        self.make_token.clone()
-                    }
-                    """,
-                    RuntimeType.IdempotencyToken,
-                )
+                if (runtimeMode.defaultToOrchestrator) {
+                    rustTemplate(
+                        """
+                        /// Returns a copy of the idempotency token provider.
+                        /// If a random token provider was configured,
+                        /// a newly-randomized token provider will be returned.
+                        pub fn make_token(&self) -> #{IdempotencyTokenProvider} {
```

Comment:
I guess it was named this before, but it's kind of weird that this is called `make_token`.

Reply:
Yeah, it's good to be renamed. Made it `token_provider` in 89185fa.

Reply:
Actually, we should make it something like `idempotency_token_provider` since `token_provider` will be used for token based auth later in the SDK.

Reply:
Will take care of it in its own PR since it needs be accompanied by a changelog entry and we want that entry to refer to a small, dedicated PR, but not this large PR 😅